### DISCUSSION TITLE
Name error

### DISCUSSION THREAD

**Discussion by allenmoench** [earthlab-education/Earth-Analytics-AY24#584] [Q&A] [answered]
Something weird going on with this code cell. It was giving me a "NameError", saying that bounds_gdf is not defined. I tried resolving this by defining bounds_gdf (bounds_gdf = denver_redlining_gdf) although I'm not sure if this violates the interoperability we're supposed to be writing into our code. It did fix the error, but now I'm getting another "NameError" saying that 'da' is not defined. That's confusing because it seems to have been defined on line 20 (da = rxr.open_rasterio(uri, mask_and_scale=True).squeeze()). Anyone know why it's doing this? 
![image](https://github.com/user-attachments/assets/5dd93041-0d6d-4104-8a52-846e4861b618)

**Comment by eculler** [comment, marked as answer]
Hi @allenmoench, it looks to me like you have a bit of an indentation problem here. Remember that indentation is how Python knows whether something is inside or outside a function -- and also that variables defined inside a function are not available outside! So, in this case, bounds_gdf is being referenced outside the function (it's unindented), but it's only defined at all because it is listed as a function parameter. Outside the function, it's called denver_redlining_gdf.

Same think with da -- you define that variable inside the function, not outside, but the code referencing it is outside.

Good news is, I think your code will work if you indent lines 21-31 one more step!

**Comment by allenmoench** [comment]
Got it to work, thanks!! It was the indentation.
